libbeat/common/mapstr.go

```diff
@@ -138,6 +138,15 @@ func (m MapStr) Put(key string, value interface{}) (interface{}, error) {
 	return walkMap(key, m, mapStrOperation{putOperation{value}, true})
 }
 
+// SetValue dereferences a value if it is a pointer. Sets the dereferenced value if not nil.
```

Comment:
Of course, I was not aware of having this normalization method in an event. Thank you for pointing this out, it will save ourselves some trouble. 
I will have a closer look and create a PR to remove empty lists then. 
@andrewkroh , @ruflin could one of you revert the commit in the beats project, I don't want to mess with your master.